### PR TITLE
Fix when mark as completed doesn't capture transaction

### PR DESCRIPTION
**Pull request by andersem**

### CODE REVIEW

File: includes/class-wc-dintero-hp-checkout.php [hunk 1]
```diff
@@ -2789,134 +2729,159 @@ public function process_refund( $order_id, $amount = null, $reason = '' ) {
...
-						'amount'      => $item_line_total_amount,
-						'line_id'     => $line_id
-					);
-					array_push( $items, $item );
-				}
+		if ($this->extract('status', $transaction) !== 'AUTHORIZED'
+			|| $this->extract('amount', $transaction) != $order_total_amount
+		) {
+			$order->add_order_note(__(
+				'Could not capture transaction: Transaction status is wrong or order and transaction amounts do not match.'
```
Comment (andersem) [inline includes/class-wc-dintero-hp-checkout.php:2816]: Would help if the note mentioned the transaction status and the amounts of both order and transaction

Comment (andersem) [inline includes/class-wc-dintero-hp-checkout.php:2816]: Would help if the note mentioned the transaction status and the amounts of both order and transaction

File: includes/class-wc-dintero-hp-checkout.php [hunk 2]
```diff
@@ -2789,134 +2729,159 @@ public function process_refund( $order_id, $amount = null, $reason = '' ) {
...
 	}
 
 	/**
 	 * Capture Payment.
 	 */
-	public function capture( $order, $transaction = null ) {
-		if ( ! empty( $order ) &&
-			 $order instanceof WC_Order &&
-			 $order->get_transaction_id() ) {
+	public function capture( $order, $transaction = null )
```
Comment (andersem) [inline includes/class-wc-dintero-hp-checkout.php:2788]: @chaikovskyia What is the difference between this capture-function and the one here: https://github.com/Dintero/Dintero.Checkout.WooCommerce/blob/0cb55c0b57f427fbd95246aec586939b25685bde/includes/class-wc-gateway-dintero-hp.php#L1400

File: includes/class-wc-dintero-hp-checkout.php [hunk 3]
```diff
@@ -1583,13 +1598,12 @@ public function get_shipping_amount() {
 		}
 		return intval(number_format(
 			WC()->cart->shipping_total + WC()->cart->shipping_tax_total,
-			wc_get_price_decimals(),
+			min(array(2, wc_get_price_decimals())),
```
[the remark this answers is not in the document]
Comment (chaikovskyia) [reply includes/class-wc-dintero-hp-checkout.php:1601]: @andersem , dintero supports only itegers right? So if we multiply a number with more than 2 decimals by 100 we will get a float...

File: includes/class-wc-dintero-hp-checkout.php (hunk 2 above)
Comment (chaikovskyia) [reply includes/class-wc-dintero-hp-checkout.php:2788]: @andersem , most likely that is a duplication, but need to check in more details and move it to a separate class.

File: includes/class-wc-dintero-hp-checkout.php (hunk 1 above)
Comment (chaikovskyia) [reply includes/class-wc-dintero-hp-checkout.php:2816]: @andersem , ok, will correct that.

Comment (chaikovskyia) [reply includes/class-wc-dintero-hp-checkout.php:2816]: @andersem , ok, will correct that.

File: includes/class-wc-dintero-hp-checkout.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (andersem) [reply includes/class-wc-dintero-hp-checkout.php:1601]: Not really, it depends on the currency, so we have a mapping from currency to number of decimals. I don't think we need to support this in woocommerce yet, though, just mentioning it.

**Pull request merged**